Cleanup errors in javadocs

sciwhiz12 opened the pull request:
This PR aims to perform the first step of cleanup for the javadocs, by removing all errors (usually caused by references to non-existent classes or inaccessible members) in javadocs. It is not a goal of this PR to rewrite, reword, add to, or improve the existing javadocs except when doing so is required to eliminate the error.

sciwhiz12 committed:
Cleanup errors in net/minecraftforge/client/event

sciwhiz12 committed:
Cleanup errors in net/minecraftforge/client/extensions

sciwhiz12 committed:
Cleanup errors in net/minecraftforge/client/model/animation

sciwhiz12 committed:
Cleanup errors in net/minecraftforge/client/model/data

sciwhiz12 committed:
Cleanup errors in net/minecraftforge/client/model/geometry

sciwhiz12 committed:
Cleanup errors in net/minecraftforge/client/model

sciwhiz12 committed:
Cleanup errors in net/minecraftforge/client/settings

sciwhiz12 committed:
Cleanup errors in net/minecraftforge/client

sciwhiz12 committed:
Cleanup errors in net/minecraftforge/common/capabilities

sciwhiz12 committed:
Cleanup errors in net/minecraftforge/common/command

sciwhiz12 committed:
Cleanup errors in net/minecraftforge/common/crafting

sciwhiz12 committed:
Cleanup errors in net/minecraftforge/common/data

sciwhiz12 committed:
Cleanup errors in net/minecraftforge/common/extensions

sciwhiz12 committed:
Cleanup errors in net/minecraftforge/common/loot

sciwhiz12 committed:
Cleanup errors in net/minecraftforge/common/util

sciwhiz12 committed:
Cleanup errors in net/minecraftforge/common

sciwhiz12 committed:
Cleanup errors in net/minecraftforge/event/brewing

sciwhiz12 committed:
Cleanup errors in net/minecraftforge/event/entity/living

ChampionAsh5357 commented:
> by removing all errors (usually caused by references to non-existent classes or inaccessible members) in javadocs

What would be the future specification when it comes to referencing private members then? Or would it just be considered bad practice to do so? My thoughts on this are mixed because sometimes we link to private members to give a general sense how a particular piece of code is used. On the other hand, you could just follow the calls to find the location of usage which may be less clear and require more thinking to understand placement.

sciwhiz12 commented:
I haven't fully tested my working theory yet on possibly resolving it, but I will state that the reason I am working to remove all of those references to inaccessible members is because both `javadoc` and IntelliJ IDEA by default do not like references to inaccessible members (at least, IDEA displays them the same as compile-time errors even if they do not impede compilation).

While I know the inspection in IDEA can be made to ignore inaccessibility for references in javadocs, and `javadoc` to my knowledge has the `-private` flag, I still need to check how Eclipse handles these references in javadocs and see if the `-private` flag actually resolves the problem. But I defer that checking after I do this error removal pass of the javadocs.

sciwhiz12 committed:
Cleanup errors in net/minecraftforge/event/entity/player

sciwhiz12 committed:
Cleanup errors in net/minecraftforge/event/entity

sciwhiz12 committed:
Cleanup errors in net/minecraftforge/event/furnace

sciwhiz12 committed:
Cleanup errors in net/minecraftforge/event/village

sciwhiz12 committed:
Cleanup errors in net/minecraftforge/event/world

sciwhiz12 committed:
Cleanup errors in net/minecraftforge/event

sciwhiz12 committed:
Cleanup errors in net/minecraftforge/fluids/capability/wrappers

sciwhiz12 committed:
Cleanup errors in net/minecraftforge/fluids

sciwhiz12 committed:
Cleanup errors in net/minecraftforge/fmllegacy/network

sciwhiz12 committed:
Cleanup errors in net/minecraftforge/fmlserverevents

sciwhiz12 committed:
Cleanup errors in net/minecraftforge/items/wrapper

sciwhiz12 committed:
Cleanup errors in net/minecraftforge/items

sciwhiz12 committed:
Cleanup errors in net/minecraftforge/fml for fmlcore

sciwhiz12 committed:
Cleanup errors in net/minecraftforge/fml/common for javafmllanguage

sciwhiz12 committed:
Cleanup errors in net/minecraftforge/fml/javafmlmod

sciwhiz12 committed:
Cleanup errors in net/minecraftforge/fml/event/lifecycle

sciwhiz12 committed:
Cleanup remaining errors in Forge packages

sciwhiz12 committed:
Cleanup errors in various patches

sciwhiz12 committed:
Remove deprecation for ResourceManagerReloadListener

Due to the resource loading system of Minecraft now using
CompletableFutures, it would've been a nightmarish task to reimplement
ISelectiveResourceReloadListener, so it was removed.

Closes ForgeForce/MinecraftForge#4

sciwhiz12 committed:
Add JDK-specific tags to javadoc tasks

Since these are used within existing javadocs, it would be less work and
beneficial to add them as recognized tags instead of rewriting all current
occurrences as regular paragraphs.

sciwhiz12 committed:
Fix errors raised by the javadoc tool

sciwhiz12 committed:
Prevent warnings from missing javadoc comments

sciwhiz12 commented:
Confirmed that no errors exist by running the following command:
```
./gradlew javadoc --console=plain --rerun-tasks -x :fmlonly:javadoc
```

ChampionAsh5357 commented on fmlcore/src/main/java/net/minecraftforge/fml/OptionalMod.java:
```diff
@@ -129,7 +129,7 @@ public void ifPresent(Consumer<? super T> consumer) {
      * result.  Otherwise return an empty {@code Optional}.
      *
-     * @apiNote This method supports post-processing on optional values, without
+     * API Note: This method supports post-processing on optional values, without
```
What's the point in doing this if you added support for the annotation? Also, if continuing to do this, should add `<p>`.

sciwhiz12 replied:
This change was done before I re-added support for those JDK-specific annoatations (after I realized that those were used a lot more often than I thought). I'll be reverting that.

ChampionAsh5357 commented on fmlcore/src/main/java/net/minecraftforge/fml/OptionalMod.java:
```diff
@@ -203,7 +203,7 @@ public T orElseGet(Supplier<? extends T> other) {
      * to be created by the provided supplier.
      *
-     * @apiNote A method reference to the exception constructor with an empty
+     * API Note: A method reference to the exception constructor with an empty
```
What's the point in doing this if you added support for the annotation? Also, if continuing to do this, should add `<p>`.

sciwhiz12 replied:
This change was done before I re-added support for those JDK-specific annoatations (after I realized that those were used a lot more often than I thought). I'll be reverting that.

ChampionAsh5357 commented on javafmllanguage/src/main/java/net/minecraftforge/fml/common/Mod.java:
```diff
@@ -84,7 +84,7 @@
             /**
              * The main Forge Event Bus.
-             * @see MinecraftForge#EVENT_BUS
+             * See {@code MinecraftForge#EVENT_BUS}
```
Should have `<p>` prefix. Though it may be better to add another annotation like `@seeplain` that doesn't resolve the symbol.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/client/event/ClientChatEvent.java:
```diff
@@ -20,13 +20,15 @@
  * ClientChatEvent is fired whenever the client is about to send a chat message or command to the server. <br>
  * This event is fired via {@link ForgeEventFactory#onClientSendMessage(String)},
- * which is executed by {@link GuiScreen#sendChatMessage(String, boolean)}<br>
+ * which is executed by {@link net.minecraft.client.gui.screens.Screen#sendMessage(String, boolean)}<br>
```
Needs to be consistency whether you should import to link or not.

sciwhiz12 replied:
What would be your recommendation? I'm of the opinion to use the imports myself, partly due to preference and keeping the javadocs small.

sciwhiz12 replied:
Went with using imports.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/client/event/GuiScreenEvent.java:
```diff
@@ -109,7 +109,7 @@ public Pre(Screen gui, List<GuiEventListener> list, Consumer<GuiEventListener> a
 
         /**
-         * This event fires right after {@link GuiScreen#initGui()}.
+         * This event fires right after the {@code init()} method in {@link Screen}.
```
I would probably make this `#init()` or `{@code Screen#init()}`.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/client/event/GuiScreenEvent.java:
```diff
@@ -218,8 +218,8 @@ public PoseStack getMatrixStack()
     /**
-     * This event fires in {@link InventoryEffectRenderer#updateActivePotionEffects()}
-     * when potion effects are active and the gui wants to move over.
+     * This event fires in {@link net.minecraft.client.gui.screens.inventory.EffectRenderingInventoryScreen} in the
```
Import inconsistency.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/client/event/GuiScreenEvent.java:
```diff
@@ -218,8 +218,8 @@ public PoseStack getMatrixStack()
-     * This event fires in {@link InventoryEffectRenderer#updateActivePotionEffects()}
-     * when potion effects are active and the gui wants to move over.
+     * This event fires in {@link net.minecraft.client.gui.screens.inventory.EffectRenderingInventoryScreen} in the
+     * {@code checkEffectRendering} method when potion effects are active and the gui wants to move over.
```
Inconsistency in formatting, either with or without method parameters.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/client/event/GuiScreenEvent.java:
```diff
@@ -463,7 +463,7 @@ public int getKeyCode()
         /**
          * Platform-specific scan code.
-         * Used for {@link InputMappings#getInputByCode(int, int)}
+         * Used for {@link com.mojang.blaze3d.platform.InputConstants#getKey(int, int)}
```
Import inconsistency.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/client/event/InputEvent.java:
```diff
@@ -215,7 +216,7 @@ public int getKey()
         /**
          * Platform-specific scan code.
-         * Used for {@link InputMappings#getInputByCode(int, int)}
+         * Used for {@link com.mojang.blaze3d.platform.InputConstants#getKey(int, int)}
```
Import inconsistency.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/client/model/ItemLayerModel.java:
```diff
@@ -53,9 +53,9 @@
 
 /**
- * Forge reimplementation of vanilla {@link ItemModelGenerator}, i.e. builtin/generated models,
+ * Forge reimplementation of vanilla {@link net.minecraft.client.renderer.block.model.ItemModelGenerator}, i.e. builtin/generated models,
```
Import inconsistency.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/client/model/ItemTextureQuadConverter.java:
```diff
@@ -44,7 +44,7 @@ private ItemTextureQuadConverter()
      * The conversion is done by scanning the texture horizontally and vertically and creating "strips" of the texture.
      * Strips that are of the same size and follow each other are converted into one bigger quad.
-     * </br>
+     * <br/>
```
Sure, though I still believe this should be stripped and done using `<p>`, though that's an out of scope comment for this PR.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/client/model/ModelLoaderRegistry.java:
```diff
@@ -104,7 +104,7 @@ public static void onModelLoadingStart()
     /**
      * Makes system aware of your loader.
-     * <b>Must be called from within {@link ModelRegistryEvent}</b>
+     * <b>Must be called from within {@link net.minecraftforge.client.event.ModelRegistryEvent}</b>
```
Import inconsistency.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/client/settings/KeyConflictContext.java:
```diff
@@ -42,7 +42,7 @@ public boolean conflicts(IKeyConflictContext other)
 
     /**
-     * Gui key bindings are only used when a {@link GuiScreen} is open.
+     * Gui key bindings are only used when a {@link net.minecraft.client.gui.screens.Screen} is open.
```
Import inconsistency.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/client/settings/KeyConflictContext.java:
```diff
@@ -59,7 +59,7 @@ public boolean conflicts(IKeyConflictContext other)
 
     /**
-     * In-game key bindings are only used when a {@link GuiScreen} is not open.
+     * In-game key bindings are only used when a {@link net.minecraft.client.gui.screens.Screen} is not open.
```
Import inconsistency.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/common/PlantType.java:
```diff
@@ -38,11 +43,11 @@
      * <p>If your new plant grows on blocks like any one of them above, never create a new {@link PlantType}.
      * This Type is only functioning in
-     * {@link net.minecraft.block.Block#canSustainPlant(BlockState, IBlockReader, BlockPos, Direction, IPlantable)},
+     * {@link net.minecraft.world.level.block.Block#canSustainPlant(BlockState, BlockGetter, BlockPos, Direction, IPlantable)},
```
Import inconsistency.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/common/command/EntitySelectorManager.java:
```diff
@@ -67,7 +67,7 @@ public static void register(String token, IEntitySelectorType type)
      *
      * If the REGISTRY does not contain a custom selector for the command being parsed,
-     * this method returns {@code null} and the vanilla logic in {@link EntitySelectorParser#parseSelector} is used.
+     * this method returns {@code null} and the vanilla logic in {@code EntitySelectorParser#parseSelector()} is used.
```
Inconsistency in code formatting, either fully qualified or simple.

sciwhiz12 replied:
What do you mean?

ChampionAsh5357 replied:
So, you choose at times to use the fully qualified name or simple names in `@code` tags arbitrarily. The same applies to `@link` tags. It should standardized across all projects on what to do for this. Though, since this is out of scope, it could probably just be an issue for now.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/common/command/EntitySelectorManager.java:
```diff
@@ -88,7 +88,7 @@ public static EntitySelector parseSelector(EntitySelectorParser parser) throws C
 
     /**
-     * This method is called in {@link EntitySelectorParser#fillSelectorSuggestions}
+     * This method is called in {@code EntitySelectorParser#fillSelectorSuggestions(SuggestionsBuilder)}
```
Inconsistency in code formatting, either fully qualified or simple.

sciwhiz12 replied:
What do you mean?

ChampionAsh5357 commented on src/main/java/net/minecraftforge/common/data/GlobalLootModifierProvider.java:
```diff
@@ -103,7 +103,6 @@ public void run(HashCache cache) throws IOException
      *
      * @param modifier      The name of the modifier, which will be the file name.
      * @param serializer    The serializer of this modifier.
-     * @param conditions    The loot conditions before {@link LootModifier#doApply} is called.
```
No `instance` replacement?

sciwhiz12 replied:
I consider adding that out of scope of this PR, as this PR is chiefly concerned with removing errors which prevent the `javadoc` tool from working.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/common/extensions/IForgeBlock.java:
```diff
@@ -476,7 +476,7 @@ default SoundType getSoundType(BlockState state, LevelReader world, BlockPos pos
     /**
      * Used to determine the state 'viewed' by an entity (see
-     * {@link ActiveRenderInfo#getBlockStateAtEntityViewpoint(World, Entity, float)}).
+     * {@link net.minecraft.client.Camera#getBlockAtCamera()}).
```
Import inconsistency.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/common/extensions/IForgeBlockEntity.java:
```diff
@@ -169,7 +171,7 @@ default void requestModelDataUpdate()
     /**
      * Allows you to return additional model data.
-     * This data can be used to provide additional functionality in your {@link net.minecraft.client.renderer.model.IBakedModel}
+     * This data can be used to provide additional functionality in your {@link net.minecraft.client.resources.model.BakedModel}
```
Import inconsistency.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/common/extensions/IForgeBlockState.java:
```diff
@@ -61,9 +61,9 @@ private BlockState self()
-     * for {@link net.minecraft.entity.LivingEntity}, this is {@code .91}.
-     * {@link net.minecraft.entity.item.ItemEntity} uses {@code .98}, and
-     * {@link net.minecraft.entity.projectile.FishingBobberEntity} uses {@code .92}.
+     * for {@link net.minecraft.world.entity.LivingEntity}, this is {@code .91}.
+     * {@link net.minecraft.world.entity.item.ItemEntity} uses {@code .98}, and
+     * {@link net.minecraft.world.entity.projectile.FishingHook} uses {@code .92}.
```
Import inconsistencies.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/common/extensions/IForgeEntity.java:
```diff
@@ -173,7 +174,7 @@ default MobCategory getClassification(boolean forSpawnCount)
      * This is used to specify that your entity has multiple individual parts, such as the Vanilla Ender Dragon.
      *
-     * See {@link net.minecraft.entity.boss.dragon.EnderDragonEntity} for an example implementation.
+     * See {@link net.minecraft.world.entity.boss.enderdragon.EnderDragon} for an example implementation.
```
Import inconsistency.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/common/extensions/IForgeEntity.java:
```diff
@@ -190,7 +191,7 @@ default boolean isMultipartEntity()
      * Only used if {@link #isMultipartEntity()} returns true.
      *
-     * See {@link net.minecraft.entity.boss.dragon.EnderDragonEntity} for an example implementation.
+     * See {@link net.minecraft.world.entity.boss.enderdragon.EnderDragon} for an example implementation.
```
Import inconsistency.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/common/extensions/IForgeItem.java:
```diff
@@ -570,7 +570,7 @@ default int getItemEnchantability(ItemStack stack)
      * called when retrieving the list of possible enchantments for an item.
      * Enchantments may additionally (or exclusively) be doing their own checks in
-     * {@link net.minecraft.enchantment.Enchantment#canApplyAtEnchantingTable(ItemStack)};
+     * {@link net.minecraft.world.item.enchantment.Enchantment#canApplyAtEnchantingTable(ItemStack)};
```
Import inconsistency.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/common/extensions/IForgeItem.java:
```diff
@@ -639,7 +639,7 @@ default boolean canContinueUsing(ItemStack oldStack, ItemStack newStack)
      * @param itemStack the ItemStack to check
      * @return the Mod ID for the ItemStack, or null when there is no specially
-     *         associated mod and {@link #getRegistryName()} would return null.
+     *         associated mod and {@link net.minecraftforge.registries.IForgeRegistryEntry#getRegistryName()} would return null.
```
Import inconsistency.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/common/extensions/IForgeItem.java:
```diff
@@ -715,7 +715,7 @@ default int getBurnTime(ItemStack itemStack, @Nullable RecipeType<?> recipeType)
 
     /**
-     * Called every tick from {@link EntityHorse#onUpdate()} on the item in the
+     * Called every tick from {@code Horse#playGallopSound(SoundEvent)} on the item in the
```
Code inconsistency.

sciwhiz12 replied:
What do you mean?

ChampionAsh5357 commented on src/main/java/net/minecraftforge/common/extensions/IForgeItemStack.java:
```diff
@@ -157,7 +156,7 @@ default boolean shouldCauseBlockBreakReset(ItemStack newStack)
      * called when retrieving the list of possible enchantments for an item.
      * Enchantments may additionally (or exclusively) be doing their own checks in
-     * {@link net.minecraft.enchantment.Enchantment#canApplyAtEnchantingTable(ItemStack)};
+     * {@link net.minecraft.world.item.enchantment.Enchantment#canApplyAtEnchantingTable(ItemStack)};
```
Import inconsistency.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/common/util/Constants.java:
```diff
@@ -113,12 +125,12 @@
 
         /**
-         * {@code data} is the {@link net.minecraft.block.Block#getStateId state id} of the block broken
+         * {@code data} is the {@link net.minecraft.world.level.block.Block#getId(BlockState)}  state id} of the block broken
```
Import inconsistency.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/common/util/ITeleporter.java:
```diff
@@ -57,7 +57,7 @@
      * @param repositionEntity a function to reposition the entity, which returns the new entity in the new dimension. This is the vanilla implementation of the dimension travel logic. If the supplied boolean is true, it is attempted to spawn a new portal.
      *
-     * @return the entity in the new World. Vanilla creates for most {@link Entity}s a new instance and copy the data. But <b>you are not allowed</b> to create a new instance for {@link PlayerEntity}s! Move the player and update its state, see {@link ServerPlayerEntity#changeDimension(net.minecraft.world.server.ServerWorld, ITeleporter)}
+     * @return the entity in the new World. Vanilla creates for most {@link Entity}s a new instance and copy the data. But <b>you are not allowed</b> to create a new instance for {@link net.minecraft.world.entity.player.Player}s! Move the player and update its state, see {@link ServerPlayer#changeDimension(ServerLevel, ITeleporter)}
```
Import inconsistency.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/common/util/LazyOptional.java:
```diff
@@ -91,7 +92,7 @@
      * This method hides an unchecked cast to the inferred type. Only use this if
      * you are sure the type should match. For capabilities, generally
-     * {@link Capability#orEmpty(Capability, LazyOptional)} should be used.
+     * {@link net.minecraftforge.common.capabilities.Capability#orEmpty(Capability, LazyOptional)} should be used.
```
Import inconsistency.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/event/RegisterCommandsEvent.java:
```diff
@@ -20,15 +20,16 @@
 /**
- * Commands are rebuilt whenever {@link DataPackRegistries} is recreated.
- * You can use this event to register your commands whenever the {@link net.minecraft.command.Commands} class in constructed.
+ * Commands are rebuilt whenever {@link net.minecraft.server.ServerResources} is recreated.
```
Import inconsistency.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/event/RegistryEvent.java:
```diff
@@ -42,7 +42,7 @@
     }
     /**
-     * Register new registries when you receive this event, through the {@link RecipeBuilder}
+     * Register new registries when you receive this event, through the {@link net.minecraftforge.registries.RegistryBuilder}
```
Import inconsistency.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/event/entity/EntityMobGriefingEvent.java:
```diff
@@ -20,16 +20,18 @@
  * <li>{@link Result#ALLOW} means this instance of mob griefing is allowed.</li>
  * <li>{@link Result#DEFAULT} means the {@code mobGriefing} game rule is used to determine the behaviour.</li>
- * <li>{@link Result#DENY} means this instance of mob griefing is not allowed.</li><br>
+ * <li>{@link Result#DENY} means this instance of mob griefing is not allowed.</li></ul>
```
Shouldn't ending tag be on a separate line for proper formatting?

ChampionAsh5357 commented on src/main/java/net/minecraftforge/event/entity/EntityTeleportEvent.java:
```diff
@@ -64,7 +65,7 @@ public EntityTeleportEvent(Entity entity, double targetX, double targetY, double
     /**
      * EntityTeleportEvent.TeleportCommand is fired before a living entity is teleported
-     * from use of {@link net.minecraft.command.impl.TeleportCommand}.
+     * from use of {@link net.minecraft.server.commands.TeleportCommand}.
```
Import inconsistency.

sciwhiz12 replied:
This cannot be changed to an import, as it would then conflict with the `TeleportCommand` inner class.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/event/entity/EntityTeleportEvent.java:
```diff
@@ -88,7 +89,7 @@ public TeleportCommand(Entity entity, double targetX, double targetY, double tar
     /**
      * EntityTeleportEvent.SpreadPlayersCommand is fired before a living entity is teleported
-     * from use of {@link net.minecraft.command.impl.SpreadPlayersCommand}.
+     * from use of {@link net.minecraft.server.commands.SpreadPlayersCommand}.
```
Import inconsistency.

sciwhiz12 replied:
This cannot be changed to an import, as it would then conflict with the `SpreadPlayersCommand` inner class.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/event/entity/PlaySoundAtEntityEvent.java:
```diff
@@ -22,11 +22,16 @@
  * PlaySoundAtEntityEvent is fired a sound is to be played at an Entity<br>
  * This event is fired whenever a sound is set to be played at an Entity such as in
- * {@link ClientPlayerEntity#playSound(SoundEvent, float, float)} and {@link World#playSound(PlayerEntity, double, double, double, SoundEvent, SoundCategory, float, float)}.<br>
+ * {@link net.minecraft.client.player.LocalPlayer#playSound(SoundEvent, float, float)} and
```
Import inconsistency.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/event/entity/living/BabyEntitySpawnEvent.java:
```diff
@@ -19,18 +19,21 @@
+ * Parents will have disengaged their relationship. {@link Cancelable} <br>
+ * It is possible to change the child completely by using {@link #setChild(AgeableMob)} <br>
+ * This event is fired from {@link Animal#spawnChildFromBreeding(ServerLevel, Animal)} and
+ * {@link net.minecraft.world.entity.animal.Fox#spawnChildFromBreeding(ServerLevel, Animal)} <br>
```
Import inconsistency.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/event/entity/living/LivingDamageEvent.java:
```diff
@@ -28,10 +29,10 @@
  * This event is fired whenever an Entity is damaged in
- * {@link EntityLivingBase#damageEntity(DamageSource, float)} and
- * {@link EntityPlayer#damageEntity(DamageSource, float)}.<br>
+ * {@code LivingEntity#actuallyHurt(DamageSource, float)} and
+ * {@code net.minecraft.world.entity.player.Player#actuallyHurt(DamageSource, float)}.<br>
```
Code inconsistency.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/event/entity/living/LivingDeathEvent.java:
```diff
@@ -19,18 +19,20 @@
- * {@link EntityPlayer#onDeath(DamageSource)}, and
- * {@link EntityPlayerMP#onDeath(DamageSource)}. <br>
+ * {@link LivingEntity#die(DamageSource)},
+ * {@link net.minecraft.world.entity.player.Player#die(DamageSource)}, and
+ * {@link net.minecraft.server.level.ServerPlayer#die(DamageSource)}. <br>
```
Import inconsistency.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/event/entity/living/LivingDestroyBlockEvent.java:
```diff
@@ -19,13 +19,16 @@
 
 /**
- * Fired when the ender dragon or wither attempts to destroy a block and when ever a zombie attempts to break a door. Basically a event version of {@link Block#canEntityDestroy(IBlockState, IBlockAccess, BlockPos, Entity)}<br>
+ * Fired when the ender dragon or wither attempts to destroy a block and when ever a zombie attempts to break a door. Basically a event version of {@link net.minecraft.world.level.block.Block#canEntityDestroy(BlockState, BlockGetter, BlockPos, Entity)}<br>
```
Import inconsistency.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/event/entity/living/LivingEvent.java:
```diff
@@ -70,10 +73,10 @@ public LivingEntity getEntityLiving()
      * This event is fired whenever an Entity jumps in
-     * {@link EntityLivingBase#jump()}, {@link EntityMagmaCube#jump()},
-     * and {@link EntityHorse#jump()}.<br>
+     * {@code LivingEntity#jumpFromGround()}, {@code net.minecraft.world.entity.monster.MagmaCube#jumpFromGround()},
+     * and {@code Horse#jumpFromGround()}.<br>
```
Code inconsistency.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/event/entity/living/LivingHurtEvent.java:
```diff
@@ -21,14 +21,17 @@
  * This event is fired whenever an Entity is hurt in
- * {@link EntityLivingBase#damageEntity(DamageSource, float)} and
- * {@link EntityPlayer#damageEntity(DamageSource, float)}.<br>
+ * {@code LivingEntity#actuallyHurt(DamageSource, float)} and
+ * {@code net.minecraft.world.entity.player.Player#actuallyHurt(DamageSource, float)}.<br>
```
Code inconsistency.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/event/entity/living/LivingKnockBackEvent.java:
```diff
@@ -19,20 +19,23 @@
- * {@link EntityLivingBase#blockWithShield(EntityLivingBase)},
- * {@link EntityMob#attackEntityAsMob(Entity)} and
- * {@link EntityPlayer#attackTargetEntityWithCurrentItem(Entity)} <br>
+ * {@link LivingEntity#hurt(DamageSource, float)},
+ * {@code LivingEntity#blockUsingShield(LivingEntity)},
+ * {@link net.minecraft.world.entity.Mob#doHurtTarget(Entity)} and
+ * {@link net.minecraft.world.entity.player.Player#attack(Entity)} <br>
```
Import/code inconsistency.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/event/entity/player/ArrowLooseEvent.java:
```diff
@@ -29,7 +31,7 @@
  * ArrowLooseEvent is fired when a player stops using a bow.<br>
  * This event is fired whenever a player stops using a bow in
- * {@link ItemBow#onPlayerStoppedUsing(ItemStack, World, EntityLivingBase, int)}.<br>
+ * {@link net.minecraft.world.item.BowItem#releaseUsing(ItemStack, Level, LivingEntity, int)}.<br>
```
Import inconsistency.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/event/entity/player/PlayerDestroyItemEvent.java:
```diff
@@ -19,26 +19,39 @@
+ * {@link MultiPlayerGameMode#useItem(Player, Level, InteractionHand)},
+ * {@link MultiPlayerGameMode#useItemOn(LocalPlayer, ClientLevel, InteractionHand, BlockHitResult)} ,
+ * {@link Player#attack(Entity)},
+ * {@code Player#hurtCurrentlyUsedShield(float)},
```
Code inconsistency.

sciwhiz12 replied:
In what way?

ChampionAsh5357 commented on src/main/java/net/minecraftforge/event/world/ChunkDataEvent.java:
```diff
@@ -57,7 +64,7 @@ public CompoundTag getData()
      * ChunkDataEvent.Load is fired when vanilla Minecraft attempts to load Chunk data.<br>
      * This event is fired during chunk loading in
-     * {@link net.minecraft.world.chunk.storage.ChunkSerializer.read(ServerWorld, TemplateManager, PointOfInterestManager, ChunkPos, CompoundNBT)} which means it is async, so be careful.<br>
+     * {@link net.minecraft.world.level.chunk.storage.ChunkSerializer#read(ServerLevel, StructureManager, PoiManager, ChunkPos, CompoundTag)} which means it is async, so be careful.<br>
```
Import inconsistency.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/event/world/ChunkDataEvent.java:
```diff
@@ -84,7 +91,7 @@ public Load(ChunkAccess chunk, CompoundTag data, ChunkStatus.ChunkType status)
      * ChunkDataEvent.Save is fired when vanilla Minecraft attempts to save Chunk data.<br>
      * This event is fired during chunk saving in
-     * {@link AnvilChunkLoader#saveChunk(World, Chunk)}. <br>
+     * {@code ChunkMap#save(ChunkAccess)}. <br>
```
Code inconsistency.

sciwhiz12 replied:
In what way? From what I can tell, this is the replacement for this method.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/event/world/ChunkEvent.java:
```diff
@@ -55,9 +58,9 @@ public ChunkAccess getChunk()
      * ChunkEvent.Load is fired when vanilla Minecraft attempts to load a Chunk into the world.<br>
      * This event is fired during chunk loading in <br>
-     * {@link ChunkProviderClient#loadChunk(int, int)}, <br>
+     * {@code ChunkProviderClient#loadChunk(int, int)}, <br>
```
Code inconsistency.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/event/world/PistonEvent.java:
```diff
@@ -74,7 +74,7 @@ public PistonMoveType getPistonMoveType()
     }
```
There is a duplicate `direction` definition on the constructor of this event.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/event/world/SaplingGrowTreeEvent.java:
```diff
@@ -22,13 +22,16 @@
  * SaplingGrowTreeEvent is fired when a sapling grows into a tree.<br>
  * This event is fired during sapling growth in
- * {@link BlockSapling#generateTree(World, BlockPos, IBlockState, Random)}.<br>
+ * {@link net.minecraft.world.level.block.SaplingBlock#advanceTree(ServerLevel, BlockPos, BlockState, Random)} .<br>
```
Import inconsistency.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/fluids/FluidAttributes.java:
```diff
@@ -186,7 +188,7 @@ public final boolean isLighterThanAir()
      * To preserve the intentions of vanilla, fluids that can turn lava into obsidian should vaporize.
      * This prevents players from making the nether safe with a single bucket.
-     * Based on {@link net.minecraft.item.BucketItem#tryPlaceContainedLiquid(PlayerEntity, World, BlockPos)}
+     * Based on {@link net.minecraft.world.item.BucketItem#emptyContents(Player, Level, BlockPos, BlockHitResult)}
```
Import inconsistency.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/fluids/FluidAttributes.java:
```diff
@@ -200,9 +202,9 @@ public boolean doesVaporize(BlockAndTintGetter reader, BlockPos pos, FluidStack
+     * Called instead of placing the fluid block if {@link DimensionType#ultraWarm()} and {@link #doesVaporize(BlockAndTintGetter, BlockPos, FluidStack)} are true.
      * Override this to make your explosive liquid blow up instead of the default smoke, etc.
-     * Based on {@link net.minecraft.item.BucketItem#tryPlaceContainedLiquid(PlayerEntity, World, BlockPos)}
+     * Based on {@link net.minecraft.world.item.BucketItem#emptyContents(Player, Level, BlockPos, BlockHitResult)}
```
Import inconsistency.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/fluids/FluidUtil.java:
```diff
@@ -542,10 +542,10 @@ public static FluidActionResult tryPlaceFluid(@Nullable Player player, Level wor
      * Checks if water-like fluids should vaporize like in the nether.
      *
-     * Modeled after {@link net.minecraft.item.BucketItem#tryPlaceContainedLiquid(PlayerEntity, World, BlockPos, BlockRayTraceResult)}
+     * Modeled after {@link net.minecraft.world.item.BucketItem#emptyContents(Player, Level, BlockPos, BlockHitResult)}
```
Import inconsistency.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/fluids/FluidUtil.java:
```diff
@@ -629,9 +629,9 @@ private static IFluidHandler getFluidBlockHandler(Fluid fluid, Level world, Bloc
     /**
      * Destroys a block when a fluid is placed in the same position.
-     * Modeled after {@link net.minecraft.item.BucketItem#tryPlaceContainedLiquid(PlayerEntity, World, BlockPos, BlockRayTraceResult)}
+     * Modeled after {@link net.minecraft.world.item.BucketItem#emptyContents(Player, Level, BlockPos, BlockHitResult)}
```
Import inconsistency.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/fmllegacy/network/FMLHandshakeHandler.java:
```diff
@@ -47,24 +49,24 @@
  *
- * <p>An instance is created during {@link net.minecraft.network.handshake.client.CHandshakePacket} handling, and attached
- * to the {@link NetworkManager#channel} via {@link FMLNetworkConstants#FML_HANDSHAKE_HANDLER}.
+ * <p>An instance is created during {@link net.minecraft.network.protocol.handshake.ClientIntentionPacket} handling, and attached
```
Import inconsistency.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/fmllegacy/network/FMLHandshakeHandler.java:
```diff
@@ -47,24 +49,24 @@
- * <p>The {@link #loginWrapper} transforms these messages into {@link net.minecraft.network.login.client.CCustomPayloadLoginPacket}
- * and {@link net.minecraft.network.login.server.SCustomPayloadLoginPacket} compatible messages, by means of wrapping.
+ * <p>The {@link #loginWrapper} transforms these messages into {@link net.minecraft.network.protocol.login.ServerboundCustomQueryPacket}
+ * and {@link net.minecraft.network.protocol.login.ClientboundCustomQueryPacket} compatible messages, by means of wrapping.
```
Import inconsistency.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/fmllegacy/network/FMLNetworkConstants.java:
```diff
@@ -50,7 +50,7 @@
     static final List<EventNetworkChannel> mcRegChannels = NetworkInitialization.buildMCRegistrationChannels();
     /**
-     * Return this value in your {@link net.minecraftforge.fml.ExtensionPoint#DISPLAYTEST} function to be ignored.
+     * Return this value in your {@link net.minecraftforge.fml.IExtensionPoint.DisplayTest} function to be ignored.
```
Import inconsistency.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/fmllegacy/network/FMLPlayMessages.java:
```diff
@@ -45,7 +45,7 @@
     /**
      * Used to spawn a custom entity without the same restrictions as
-     * {@link net.minecraft.network.play.server.SSpawnObjectPacket} or {@link net.minecraft.network.play.server.SSpawnMobPacket}
+     * {@link net.minecraft.network.protocol.game.ClientboundAddEntityPacket} or {@link net.minecraft.network.protocol.game.ClientboundAddMobPacket}
```
Import inconsistency.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/fmllegacy/network/NetworkHooks.java:
```diff
@@ -158,13 +158,9 @@ public static boolean tickNegotiation(ServerLoginPacketListenerImpl netHandlerLo
      * Request to open a GUI on the client, from the server
      *
-     * Refer to {@link net.minecraftforge.fml.ExtensionPoint#CONFIGGUIFACTORY} for how to provide a function to consume
+     * Refer to {@link net.minecraftforge.fml.IExtensionPoint.DisplayTest} for how to provide a function to consume
```
Import inconsistency.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/fmllegacy/network/NetworkHooks.java:
```diff
@@ -176,13 +172,9 @@ public static void openGui(ServerPlayer player, MenuProvider containerSupplier)
      * Request to open a GUI on the client, from the server
      *
-     * Refer to {@link net.minecraftforge.fml.ExtensionPoint#CONFIGGUIFACTORY} for how to provide a function to consume
+     * Refer to {@link net.minecraftforge.fml.IExtensionPoint.DisplayTest} for how to provide a function to consume
```
Import inconsistency.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/fmllegacy/network/NetworkHooks.java:
```diff
@@ -194,12 +186,9 @@ public static void openGui(ServerPlayer player, MenuProvider containerSupplier,
      * Request to open a GUI on the client, from the server
      *
-     * Refer to {@link net.minecraftforge.fml.ExtensionPoint#CONFIGGUIFACTORY} for how to provide a function to consume
+     * Refer to {@link net.minecraftforge.fml.IExtensionPoint.DisplayTest} for how to provide a function to consume
```
Import inconsistency.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/fmllegacy/network/NetworkRegistry.java:
```diff
@@ -256,7 +257,7 @@ private static boolean validateChannels(final Map<ResourceLocation, String> inco
 
     /**
-     * Retrieve the {@link LoginPayload} list for dispatch during {@link FMLHandshakeHandler#tickLogin(NetworkManager)} handling.
+     * Retrieve the {@link LoginPayload} list for dispatch during {@code FMLHandshakeHandler#tickLogin(Connection)} handling.
```
Code inconsistency.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/fmlserverevents/FMLServerAboutToStartEvent.java:
```diff
@@ -22,7 +22,7 @@
 
 /**
- * Called before the server begins loading anything. Called after {@link InterModProcessEvent} on the dedicated
+ * Called before the server begins loading anything. Called after {@link net.minecraftforge.fml.event.lifecycle.InterModProcessEvent} on the dedicated
```
Import inconsistency.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/items/IItemHandler.java:
```diff
@@ -97,7 +100,7 @@
     /**
      * <p>
-     * This function re-implements the vanilla function {@link IInventory#isItemValidForSlot(int, ItemStack)}.
+     * This function re-implements the vanilla function {@link net.minecraft.world.Container#canPlaceItem(int, ItemStack)}.
```
Import inconsistency.

ChampionAsh5357 commented on src/main/java/net/minecraftforge/registries/ForgeRegistries.java:
```diff
@@ -62,7 +62,7 @@
  * A class that exposes static references to all vanilla and Forge registries.
  * Created to have a central place to access the registries directly if modders need.
- * It is still advised that if you are registering things to go through {@link GameRegistry} register methods, but queries and iterations can use this.
+ * It is still advised that if you are registering things to go through {@link net.minecraftforge.fmllegacy.common.registry.GameRegistry} register methods, but queries and iterations can use this.
```
Import inconsistency.

ChampionAsh5357 reviewed (changes requested):
A lot of inconsistency when it comes to formatting in determining when to import or use fully qualified name. Same goes for code tags as well. There is a bunch of additional inconsistency when it comes to method reference format as well.

Also, just as an additional comment since it's probably out of scope, there really should be something done about the `<br>` formats.

sciwhiz12 committed:
Reintroduce use of `@apiNote` tag in OptionalMod

sciwhiz12 committed:
Add missing paragraph tag

sciwhiz12 committed:
Fix import, link, and code inconsistencies

sciwhiz12 committed:
Fix other inconsistencies

sciwhiz12 committed:
Fix incorrect references to extension point

Note that the javadoc is still incorrect, but correcting that factual
error is out of scope for this PR.

sciwhiz12 committed:
Remove duplicate direction parameter

sciwhiz12 committed:
Amend method reference to be more concise

sciwhiz12 committed:
Move list end tag to next line

ChampionAsh5357 reviewed (approved):
Other than whether you consider the tags out of scope, this PR is fine.

sciwhiz12 commented:
From my recollection while doing this, my process was first to make a 'valid' `{@link}` tag, ignoring accessibility -- that is, the `{@link}` tag must point to a valid class or member thereof. After which, I then change those tags which are marked invalid by IDEA due to being unaccessible at that location to be `{@code}` tags. This was to ensure that the referenced members in the javadocs did actually exist in the code, while still removing the accessibility errors posed to me by IDEA. This meant that the use of simple name or qualified names were dependent on whether a similar import already existed.

In any case, I say we consider the use of either simple names or fully qualified names in `{@code}` tags for the reason above to be equally valid, except where the use of a simple name may introduce ambiguity as to what class/member is being referenced.
